.github/workflows/push.yml

```diff
@@ -8,56 +7,89 @@ on:
       - '**'
 
 jobs:
-  lint:
+  pre-checks:
     runs-on: ubuntu-latest
     steps:
       # Install golang
       - uses: actions/setup-go@v2
         with:
-          go-version: '^1.13.1'
+          go-version: 1.14
 
-      # Setup gopath
-      - name: Setting up GOPATH 
+      # Checkout to the latest commit
+      # On specific directory/path
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: gofmt check
         run: |
-          echo "GOPATH=${GITHUB_WORKSPACE}/go" >> $GITHUB_ENV
-        
+          if [ "$(gofmt -s -l . | wc -l)" -ne 0 ]
+          then
+           echo "The following files were found to be not go formatted:"
+           gofmt -s -l .
+           echo "Please run 'make fmt' to go format the above files."
+           exit 1
+          fi
+
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v1
+
+      - name: unused-package check
+        run: |
+          make unused-package-check
+
+  image-build:
+    runs-on: ubuntu-latest
+    needs: pre-checks
+    steps:
       # Checkout to the latest commit
       # On specific directory/path
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
         with:
-          path: go/src/github.com/${{github.repository}}
+          version: latest
 
-      #TODO: Add Dockerfile linting
-      # Running go-lint
-      - name: Checking Go-Lint
-        run : |
-          sudo apt-get update && sudo apt-get install golint
+      - name: login to GitHub Container Registry
+        run: echo ${{ secrets.DPASS }} | docker login -u ${{ secrets.DNAME }} --password-stdin
+
+      - name: Build & Push Docker Image
+        env:
+          DOCKER_REPO: litmuschaos
+          DOCKER_IMAGE: chaos-exporter
+          DOCKER_TAG: ci
+          DNAME: ${{ secrets.DNAME }}
+          DPASS: ${{ secrets.DPASS }}
+        run: |
           cd go/src/github.com/${{github.repository}}
```

Comment:
Is this required here?

Reply:
yes, to do docker login.

Reply:
I have removed the build script.

Reply:
okay

Reply:
`cd go/src/github.com/${{github.repository}}` path now does not exist  this can we remove it.